os/arch/arm/src/armv7-r/arm_blocktask.c

```diff
@@ -143,13 +143,9 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
 
 			rtcb = this_task();
 
-#ifdef CONFIG_TASK_SCHED_HISTORY
-			/* Save the task name which will be scheduled */
-			save_task_scheduling_status(rtcb);
-#endif
+			/* Switch context of rtcb */
 
-			/* Then switch contexts. */
-			up_restorestate(rtcb->xcp.regs);
```

Comment:
There are many similar cases in armv7-r code like arm_schedyield.c
Please check this.

Reply:
I added missing cases. Please check again.